convert2rhel/unit_tests/pkghandler_test.py

```diff
@@ -980,6 +980,104 @@ def test_get_packages_to_update_dnf(packages, reposdir, pretend_os, monkeypatch)
+
+    # Get all enabled repos in the current system
+    base.read_all_repos()
+    enabled_repos = [repo.id for repo in base.repos.iter_enabled()]
+
+    # Check if the list is not empty and the RHEL repo is enabled
+    assert enabled_repos
+    assert rhel_repo_id in enabled_repos
+
+    # Choose one of the enabled repos on the system
+    # Avoid getting the rhel_repo_id if on 0 index, then get the repo id from 1
+    repo_for_disable = [enabled_repos[0]] if enabled_repos[0] != rhel_repo_id else [enabled_repos[1]]
+
+    # Use the previously chosen repo for disabling by the user
+    # When user add repo through --enable repo option in _get_packages_to_update_dnf should be disabled
+    monkeypatch.setattr(tool_opts, "enablerepo", repo_for_disable)
+
+    # Patch the original Dnf Base to ours to have access to it and contain the added repo
+    monkeypatch.setattr(pkgmanager, "Base", mock.Mock(return_value=base))
+
+    _get_packages_to_update_dnf(reposdir=None)
+
+    # Get enabled repos after the _get_packages_to_update_dnf where the RHEL repos are disabled
+    enabled_repos = [repo.id for repo in base.repos.iter_enabled()]
+
+    assert rhel_repo_id not in enabled_repos
+    assert repo_for_disable not in enabled_repos
+
+
+@pytest.mark.skipif(
+    pkgmanager.TYPE != "yum",
+    reason="No yum module detected on the system, skipping it.",
+)
+def test_get_packages_to_update_yum_rhel_repos(monkeypatch):
+    class MockPkgList:
+        """Class for mocking the doPackageList and having the needed attributes."""
+
+        def __init__(self, pkgnarrow):
+            self.updates = []
+            assert pkgnarrow == "updates"
+
+    # Mock unnecessary calls for testing
+    monkeypatch.setattr(pkgmanager.YumBase, "doPackageLists", value=MockPkgList)
+
+    # Create YUM Base object to have access to it
+    base = pkgmanager.YumBase()
+
+    # Add RHEL repo to the YUM Base
+    rhel_repo_id = "rhel-7-server-rpms"
+    base.repos.add(pkgmanager.repos.Repository(rhel_repo_id))
+    base.repos.enableRepo(rhel_repo_id)
+
+    # Get all enabled repos in the current system
+    enabled_repos = [repo.id for repo in base.repos.listEnabled()]
+
+    # Check if the list is not empty and the RHEL repo is enabled
+    assert enabled_repos
+    assert rhel_repo_id in enabled_repos
+
+    # Choose one of the enabled repos on the system
+    # Avoid getting the rhel_repo_id if on 0 index, then get the repo id from 1
+    repo_for_disable = [enabled_repos[0]] if enabled_repos[0] != rhel_repo_id else [enabled_repos[1]]
+
+    # Use the previously chosen repo for disabling by the user
+    # When user adds a repo through the --enable-repo option, it should be disabled
+    monkeypatch.setattr(tool_opts, "enablerepo", repo_for_disable)
+
+    # Patch the original YUM Base to ours to have access to it and contain the added repo
+    monkeypatch.setattr(pkgmanager, "YumBase", mock.Mock(return_value=base))
```

Comment:
Good approach, that means we still test with the Yum/DnfBase still and not just plainly mocking it out

Reply:
Yes, exactly